packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -352,6 +352,16 @@ describe('Context bindings - Injecting dependencies of classes', () => {
       expect(ctx.getSync(HASH_KEY)).to.equal('a-value');
     });
 
+    it('throws error if binding key is empty', async () => {
+      class InvalidStore {
+        constructor(@inject.setter('') public setter: Setter<string>) {}
```

Comment:
hmm, why the `setter` has a different behaviour than the `binding`?

Reply:
We don't want to have the `setter` changes its own binding/value - self mutation is not desired.